app/_common/components/WebSocketProvider.tsx

```diff
@@ -0,0 +1,142 @@
+"use client";
+
+import React, { useEffect, useRef, useState } from "react";
+
+import { useQueryUserData } from "@/app/my-page/_hooks/useQueryUserData";
+
+import { cn } from "../shadcn/utils";
+import { useToast } from "../shadcn/ui/use-toast";
+import { Progress } from "../shadcn/ui/progress";
+import { Avatar, AvatarFallback, AvatarImage } from "../shadcn/ui/avatar";
+import { getAchievementDetail } from "../api/achievements";
+import { ModalState } from "./types/achievementModal";
+import AchievementModal from "./AchievementModal";
+
+interface WebSocketMessage {
+  userId: number;
+  achievementId: number;
+  completed: boolean;
+  progressCount: number;
```

Comment:
이것두 Achievement["id]
```suggestion
  userId: Achievement["userId"];
  achievementId: Achievement["achievementId"];
  completed: Achievement["completed"];
  progressCount: Achievement["progressCount"];
```
api에서 id 타입 achievement에서 참조한 것처럼 이렇게하면 유지보수성이 올라갈 것 같아요 👍